pkg/transform/routes.go

```diff
@@ -83,12 +83,13 @@ func (rn *routerNode) provideRoutes() (pipe.MiddleFunc[[]request.Span, []request
 
 	return func(in <-chan []request.Span, out chan<- []request.Span) {
 		for spans := range in {
-			filtered := make([]request.Span, 0, len(spans))
 			for i := range spans {
 				s := &spans[i]
 				if ignoreEnabled {
 					if discarder.Find(s.Path) != "" {
 						if ignoreMode == IgnoreAll {
+							s.SetIgnoreMetrics()
```

Comment:
Because of this continue in the loop, which is rare, we were duplicating the spans array. I decided to let the spans make it all the way to the terminal nodes and be ignored there.

Reply:
If we keep this design, should I ignore the spans in `print traces` debug code? It might be useful to see ignored paths...